jupyter_kernel_mgmt/tests/test_kernelspec.py

```diff
@@ -24,177 +23,168 @@
-
-        self.ksm = kernelspec.KernelSpecManager()
-        self.prov_ksm = kernelspec.KernelSpecManager(kernel_file='prov_kernel.json')
-
-        td2 = TemporaryDirectory()
-        self.addCleanup(td2.cleanup)
-        self.installable_kernel = td2.name
-        with open(pjoin(self.installable_kernel, 'kernel.json'), 'w') as f:
-            json.dump(sample_kernel_json, f)
-
-    def tearDown(self):
-        self.env_patch.stop()
-
-    def test_find_kernel_specs(self):
-        kernels = self.ksm.find_kernel_specs()
-        self.assertEqual(kernels['sample'], self.sample_kernel_dir)
-        self.assertNotIn('prov_sample1', kernels)
-        self.assertNotIn('prov_sample2', kernels)
-
-    def test_get_kernel_spec(self):
-        ks = self.ksm.get_kernel_spec('SAMPLE')  # Case insensitive
-        self.assertEqual(ks.resource_dir, self.sample_kernel_dir)
-        self.assertEqual(ks.argv, sample_kernel_json['argv'])
-        self.assertEqual(ks.display_name, sample_kernel_json['display_name'])
-        self.assertEqual(ks.env, {})
-        self.assertEqual(ks.metadata, {})
-        with self.assertRaises(kernelspec.NoSuchKernel):
-            self.ksm.get_kernel_spec('prov1_sample')
-
-    def test_find_all_specs(self):
-        kernels = self.ksm.get_all_specs()
-        self.assertEqual(kernels['sample']['resource_dir'], self.sample_kernel_dir)
-        self.assertIsNotNone(kernels['sample']['spec'])
-    
-    def test_kernel_spec_priority(self):
-        td = TemporaryDirectory()
-        self.addCleanup(td.cleanup)
-        sample_kernel = install_sample_kernel(td.name)
-        self.ksm.kernel_dirs.append(td.name)
-        kernels = self.ksm.find_kernel_specs()
-        self.assertEqual(kernels['sample'], self.sample_kernel_dir)
-        self.ksm.kernel_dirs.insert(0, td.name)
-        kernels = self.ksm.find_kernel_specs()
-        self.assertEqual(kernels['sample'], sample_kernel)
-
-    def test_install_kernel_spec(self):
-        self.ksm.install_kernel_spec(self.installable_kernel,
+from .utils import install_sample_kernel, sample_kernel_json
+
+
+@pytest.fixture
+def setup_test(setup_env):
+    pytest.sample_kernel_dir = install_sample_kernel(
+        pjoin(paths.jupyter_data_dir(), 'kernels'))
+    pytest.prov_sample1_kernel_dir = install_sample_kernel(
+        pjoin(paths.jupyter_data_dir(), 'kernels'), 'prov_sample1', 'prov_kernel.json')
+    pytest.prov_sample2_kernel_dir = install_sample_kernel(
+        pjoin(paths.jupyter_data_dir(), 'kernels'), 'prov_sample2', 'prov_kernel.json')
+
+    pytest.ksm = kernelspec.KernelSpecManager()
+    pytest.prov_ksm = kernelspec.KernelSpecManager(kernel_file='prov_kernel.json')
+
+    td2 = TemporaryDirectory()
+    pytest.installable_kernel = td2.name
+    with open(pjoin(pytest.installable_kernel, 'kernel.json'), 'w') as f:
+        json.dump(sample_kernel_json, f)
+
+    yield pytest.installable_kernel
+    shutil.rmtree(pytest.installable_kernel)
```

Comment:
I think this would be better broken up into at least 2 separate fixtures, separating the installable kernel from the installed ones, and possibly also separating the different installed kernels. Of course, they should all then avoid modifying the `pytest` global. :-)

Reply:
Sorry I missed these.  Thanks.

Reply:
Yep - sorry I missed these.  Also fixing the same issue in test_client.py.